modules/nf-core/kraken2/add/main.nf

```diff
@@ -0,0 +1,38 @@
+process KRAKEN2_ADD {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' :
+        'biocontainers/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' }"
+
+    input:
+    tuple val(meta), path(fasta)
+    path taxonomy_names
+    path taxonomy_nodes
+    path accession2taxid
+
+    output:
+    tuple val(meta), path("$prefix/"), emit: db
+    path "versions.yml"              , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
```

Comment:
```suggestion
    prefix = task.ext.prefix ?: "${meta.id}"
```

that's the source of your error: if you use `def` it means that the variable can only be used within the scope it's defined (in this case `script`). Remove the `def` and you can use it in e.g. the `output:` scope.